cmake/compilerFlags.cmake

```diff
@@ -136,4 +137,10 @@ if(MSVC)
     # Object Level Parallelism
     add_compile_options(/MP)
     add_definitions(-DNOMINMAX -DWIN32_LEAN_AND_MEAN)
+    
+    # https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/
+    if (MSVC_VERSION GREATER_EQUAL "1900") # VS2019 and up

```

Comment:
I think the condition is not totally right. MSVC_VERSION 1900 == VS 14.0 == VS 2015 where the flag is not available. You can see the warnings in this Appveyor job which is using VS 2015:

https://ci.appveyor.com/project/piponazo/exiv2-wutfp/builds/32652184/job/32l5de0a9cpseo2u

According to the official cmake doc (https://cmake.org/cmake/help/v3.17/variable/MSVC_VERSION.html?highlight=msvc_version) we should be using:

```
if (MSVC_VERSION GREATER_EQUAL "1920") # VS2019 and up
```

Reply:
I will also commit that change once I see the feedback from previous changes in CI.

Reply:
You're right.  It should be 1920 for VS 2019 and later.

Reply:
I noticed that VS2017 also recognise the flag, so that I finally changed the value to 1910 :wink:

Reply:
I'm trying it at the moment on 2015/17/19.  VS is downloading/updating VS2017 and VS2019 at the moment.

Reply:
You can also check the output directly on the Appveyor jobs if you do not need to do much with the compilers:

https://ci.appveyor.com/project/piponazo/exiv2-wutfp/builds/32652433

Note that we are building just with 2013, 2015 & 2017 in there. In the past we were building with much older versions, but at some point it got so troublesome that we decided to drop some of the older ones.

Reply:
VS2017 accepts `/Zc:__cplusplus`. 

VS2015 produces the warning:
```
cl : Command line warning D9002: ignoring unknown option '/Zc:__cplusplus' [C:\Users\rmills\gnu\github\exiv2\buildserver\build\samples\convert-test.vcxproj]
```

I'll change the code in `cmake/compilerFlags.cmake` to 1910 (VS2017)

Reply:
I already changed it in this same PR robin. Please check the changes I did.